cluster-autoscaler/core/scaledown/planner/controller.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package planner
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes"
+)
+
+type controllerCalculatorImpl struct {
+	listers kubernetes.ListerRegistry
+}
+
+func newControllerReplicasCalculator(listers kubernetes.ListerRegistry) controllerReplicasCalculator {
+	return &controllerCalculatorImpl{listers: listers}
+}
+
+func (c *controllerCalculatorImpl) getReplicas(ownerRef metav1.OwnerReference, namespace string) (*replicasInfo, error) {
+	result := &replicasInfo{}
+	switch ownerRef.Kind {
+	case "StatefulSet":
+		sSet, err := c.listers.StatefulSetLister().StatefulSets(namespace).Get(ownerRef.Name)
+		if err != nil {
+			return nil, err
+		}
+		result.currentReplicas = sSet.Status.CurrentReplicas
+		result.targetReplicas = *sSet.Spec.Replicas
```

Comment:
This can cause panic when `sSet.Spec.Replicas` is `nil`. We should handle that and treat it as 1. (IIUC `nil` means value unspecified, use default). Same below. We should also have tests covering such cases.

Reply:
Added checks, defaulting & tests